### PR TITLE
Whatafix/security measures

### PR DESCRIPTION
**Pull request by Raycoms**
This fixes some incredibly incredible bugs which should never happen but do.

# Changes proposed in this pull request:
- Impossibly true if checks
- Fix barbarian despawn
- Fix barbarian spawn.
Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/event/ClientEventHandler.java [hunk 1]
```diff
@@ -101,7 +101,7 @@ public void renderWorldLastEvent(@NotNull final RenderWorldLastEvent event)
             }
 
             final ColonyView colony = ColonyManager.getClosestColonyView(world, player.getPosition());
-            if(colony != null && colony.getPermissions().hasPermission(player, Action.ACCESS_HUTS))
+            if(colony != null && player != null && colony.getPermissions().hasPermission(player, Action.ACCESS_HUTS))
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/event/ClientEventHandler.java:104]: MAJOR Remove this expression which always evaluates to "true" [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2589)

File: src/main/java/com/minecolonies/coremod/colony/permissions/Permissions.java [hunk 2]
```diff
@@ -947,7 +947,7 @@ public boolean hasPermission(final UUID id, @NotNull final Action action)
         public boolean hasPermission(final Rank rank, @NotNull final Action action)
         {
             return (rank == Rank.OWNER && action != Action.GUARDS_ATTACK)
-                     || Utils.testFlag(permissions.get(rank), action.getFlag());
+                     || (permissions != null && action != null && Utils.testFlag(permissions.get(rank), action.getFlag()));
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/permissions/Permissions.java:950]: MAJOR Remove this expression which always evaluates to "true" [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2589)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 40 issues
* ![MAJOR][MAJOR] 8 major
* ![MINOR][MINOR] 32 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BuildingLumberjack.java#L248](https://github.com/Minecolonies/minecolonies/blob/cc523f3ab2436036c902882a6afedfa90ff21ca2/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java#L248): Reduce this class from 73 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [Permissions.java#L29](https://github.com/Minecolonies/minecolonies/blob/cc523f3ab2436036c902882a6afedfa90ff21ca2/src/main/java/com/minecolonies/coremod/colony/permissions/Permissions.java#L29): class "Permissions" has 36 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Permissions.java#L325](https://github.com/Minecolonies/minecolonies/blob/cc523f3ab2436036c902882a6afedfa90ff21ca2/src/main/java/com/minecolonies/coremod/colony/permissions/Permissions.java#L325): Either log or rethrow this exception. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [Permissions.java#L786](https://github.com/Minecolonies/minecolonies/blob/cc523f3ab2436036c902882a6afedfa90ff21ca2/src/main/java/com/minecolonies/coremod/colony/permissions/Permissions.java#L786): Remove this unused private "isSubscriber" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [Permissions.java#L871](https://github.com/Minecolonies/minecolonies/blob/cc523f3ab2436036c902882a6afedfa90ff21ca2/src/main/java/com/minecolonies/coremod/colony/permissions/Permissions.java#L871): Reduce this class from 116 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MAJOR][MAJOR] [ClientEventHandler.java#L57](https://github.com/Minecolonies/minecolonies/blob/cc523f3ab2436036c902882a6afedfa90ff21ca2/src/main/java/com/minecolonies/coremod/event/ClientEventHandler.java#L57): Refactor this method to reduce its Cognitive Complexity from 26 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/cc523f3ab2436036c902882a6afedfa90ff21ca2/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/cc523f3ab2436036c902882a6afedfa90ff21ca2/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/cc523f3ab2436036c902882a6afedfa90ff21ca2/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/cc523f3ab2436036c902882a6afedfa90ff21ca2/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/entity/ai/mobs/barbarians/AbstractEntityBarbarian.java [hunk 3]
```diff
@@ -141,14 +141,19 @@ protected ResourceLocation getLootTable()
         return BarbarianSpawnUtils.getBarbarianLootTable(this);
     }
 
+    /**
+     * Should the barbs despawn.
+     * @return true if so.
+     */
+    private boolean shouldDespawn()
+    {
+        return worldTimeAtSpawn != 0 && (worldObj.getTotalWorldTime() - worldTimeAtSpawn) >= 1000;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/mobs/barbarians/AbstractEntityBarbarian.java]: MAJOR Assign this magic number 1000 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 56 issues
* ![MAJOR][MAJOR] 22 major
* ![MINOR][MINOR] 33 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/f0a09fcd447d3e7e3af7d44644daa09ced418355/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.380 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L55](https://github.com/Minecolonies/minecolonies/blob/f0a09fcd447d3e7e3af7d44644daa09ced418355/src/main/java/com/minecolonies/coremod/colony/Colony.java#L55): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 48 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L55](https://github.com/Minecolonies/minecolonies/blob/f0a09fcd447d3e7e3af7d44644daa09ced418355/src/main/java/com/minecolonies/coremod/colony/Colony.java#L55): class "Colony" has 93 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L700](https://github.com/Minecolonies/minecolonies/blob/f0a09fcd447d3e7e3af7d44644daa09ced418355/src/main/java/com/minecolonies/coremod/colony/Colony.java#L700): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L893](https://github.com/Minecolonies/minecolonies/blob/f0a09fcd447d3e7e3af7d44644daa09ced418355/src/main/java/com/minecolonies/coremod/colony/Colony.java#L893): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [Colony.java#L1215](https://github.com/Minecolonies/minecolonies/blob/f0a09fcd447d3e7e3af7d44644daa09ced418355/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1215): The Cyclomatic Complexity of this method "onWorldTick" is 19 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Colony.java#L1386](https://github.com/Minecolonies/minecolonies/blob/f0a09fcd447d3e7e3af7d44644daa09ced418355/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1386): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1388](https://github.com/Minecolonies/minecolonies/blob/f0a09fcd447d3e7e3af7d44644daa09ced418355/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1388): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1390](https://github.com/Minecolonies/minecolonies/blob/f0a09fcd447d3e7e3af7d44644daa09ced418355/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1390): Assign this magic number 128 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1509](https://github.com/Minecolonies/minecolonies/blob/f0a09fcd447d3e7e3af7d44644daa09ced418355/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1509): Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 54 issues
* ![MAJOR][MAJOR] 20 major
* ![MINOR][MINOR] 33 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/effed142b89b0903f33729cd642498971b7220d1/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.380 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L55](https://github.com/Minecolonies/minecolonies/blob/effed142b89b0903f33729cd642498971b7220d1/src/main/java/com/minecolonies/coremod/colony/Colony.java#L55): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 48 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L55](https://github.com/Minecolonies/minecolonies/blob/effed142b89b0903f33729cd642498971b7220d1/src/main/java/com/minecolonies/coremod/colony/Colony.java#L55): class "Colony" has 93 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L700](https://github.com/Minecolonies/minecolonies/blob/effed142b89b0903f33729cd642498971b7220d1/src/main/java/com/minecolonies/coremod/colony/Colony.java#L700): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L893](https://github.com/Minecolonies/minecolonies/blob/effed142b89b0903f33729cd642498971b7220d1/src/main/java/com/minecolonies/coremod/colony/Colony.java#L893): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [Colony.java#L1215](https://github.com/Minecolonies/minecolonies/blob/effed142b89b0903f33729cd642498971b7220d1/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1215): The Cyclomatic Complexity of this method "onWorldTick" is 19 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Colony.java#L1386](https://github.com/Minecolonies/minecolonies/blob/effed142b89b0903f33729cd642498971b7220d1/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1386): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1388](https://github.com/Minecolonies/minecolonies/blob/effed142b89b0903f33729cd642498971b7220d1/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1388): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1390](https://github.com/Minecolonies/minecolonies/blob/effed142b89b0903f33729cd642498971b7220d1/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1390): Assign this magic number 128 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1509](https://github.com/Minecolonies/minecolonies/blob/effed142b89b0903f33729cd642498971b7220d1/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1509): Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 54 issues
* ![MAJOR][MAJOR] 20 major
* ![MINOR][MINOR] 33 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/bc941d5af4097c7cc6d26aa35fb7c55814ab4c06/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.380 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L55](https://github.com/Minecolonies/minecolonies/blob/bc941d5af4097c7cc6d26aa35fb7c55814ab4c06/src/main/java/com/minecolonies/coremod/colony/Colony.java#L55): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 48 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L55](https://github.com/Minecolonies/minecolonies/blob/bc941d5af4097c7cc6d26aa35fb7c55814ab4c06/src/main/java/com/minecolonies/coremod/colony/Colony.java#L55): class "Colony" has 93 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L700](https://github.com/Minecolonies/minecolonies/blob/bc941d5af4097c7cc6d26aa35fb7c55814ab4c06/src/main/java/com/minecolonies/coremod/colony/Colony.java#L700): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L893](https://github.com/Minecolonies/minecolonies/blob/bc941d5af4097c7cc6d26aa35fb7c55814ab4c06/src/main/java/com/minecolonies/coremod/colony/Colony.java#L893): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [Colony.java#L1215](https://github.com/Minecolonies/minecolonies/blob/bc941d5af4097c7cc6d26aa35fb7c55814ab4c06/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1215): The Cyclomatic Complexity of this method "onWorldTick" is 19 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Colony.java#L1386](https://github.com/Minecolonies/minecolonies/blob/bc941d5af4097c7cc6d26aa35fb7c55814ab4c06/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1386): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1388](https://github.com/Minecolonies/minecolonies/blob/bc941d5af4097c7cc6d26aa35fb7c55814ab4c06/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1388): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1390](https://github.com/Minecolonies/minecolonies/blob/bc941d5af4097c7cc6d26aa35fb7c55814ab4c06/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1390): Assign this magic number 128 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1509](https://github.com/Minecolonies/minecolonies/blob/bc941d5af4097c7cc6d26aa35fb7c55814ab4c06/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1509): Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 57 issues
* ![MAJOR][MAJOR] 23 major
* ![MINOR][MINOR] 33 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/d03e15b750868e72c2e371b96dc981ca64ff8029/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.380 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L55](https://github.com/Minecolonies/minecolonies/blob/d03e15b750868e72c2e371b96dc981ca64ff8029/src/main/java/com/minecolonies/coremod/colony/Colony.java#L55): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 48 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L55](https://github.com/Minecolonies/minecolonies/blob/d03e15b750868e72c2e371b96dc981ca64ff8029/src/main/java/com/minecolonies/coremod/colony/Colony.java#L55): class "Colony" has 93 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L700](https://github.com/Minecolonies/minecolonies/blob/d03e15b750868e72c2e371b96dc981ca64ff8029/src/main/java/com/minecolonies/coremod/colony/Colony.java#L700): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L893](https://github.com/Minecolonies/minecolonies/blob/d03e15b750868e72c2e371b96dc981ca64ff8029/src/main/java/com/minecolonies/coremod/colony/Colony.java#L893): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [Colony.java#L1215](https://github.com/Minecolonies/minecolonies/blob/d03e15b750868e72c2e371b96dc981ca64ff8029/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1215): The Cyclomatic Complexity of this method "onWorldTick" is 19 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [Colony.java#L1386](https://github.com/Minecolonies/minecolonies/blob/d03e15b750868e72c2e371b96dc981ca64ff8029/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1386): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1388](https://github.com/Minecolonies/minecolonies/blob/d03e15b750868e72c2e371b96dc981ca64ff8029/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1388): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1390](https://github.com/Minecolonies/minecolonies/blob/d03e15b750868e72c2e371b96dc981ca64ff8029/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1390): Assign this magic number 128 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1509](https://github.com/Minecolonies/minecolonies/blob/d03e15b750868e72c2e371b96dc981ca64ff8029/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1509): Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/event/ClientEventHandler.java (hunk 1 above)
Comment (cltnschlosser) [reply src/main/java/com/minecolonies/coremod/event/ClientEventHandler.java:104]: We use player.getPosition() on the line before, so this sonar statement is correct.

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/event/ClientEventHandler.java:104]: I know but it never crashes at player.getPosition() it crashes inside the hasPermission =D

**Pull request merged**